components/web/ejabberd/Makefile

```diff
@@ -23,36 +23,36 @@
 # Copyright (c) 2011, 2015, Oracle and/or its affiliates. All rights reserved.
 # Copyright 2021 Andreas Grueninger, Grueninger GmbH, (grueni). All rights reserved.
 # Copyright 2022 Niklas Poslovski
+# Copyright 2024 Goetz Fischer, R-A-C. All rights reserved.
 #
 
 BUILD_BITS= 64
-
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME= ejabberd
 COMPONENT_FMRI= web/server/$(COMPONENT_NAME)
-COMPONENT_VERSION= 22.0.5
-HUMAN_VERSION= 22.05
-COMPONENT_PROJECT_URL=	http://www.ejabberd.im/
-COMPONENT_SRC= $(COMPONENT_NAME)-$(HUMAN_VERSION)
+COMPONENT_VERSION= 23.10
+HUMAN_VERSION= 23.10
+COMPONENT_PROJECT_URL= http://www.ejabberd.im/
+COMPONENT_SRC= $(COMPONENT_NAME)-$(COMPONENT_VERSION)
 COMPONENT_ARCHIVE= $(COMPONENT_SRC).tar.gz
-COMPONENT_ARCHIVE_HASH= sha256:b8e93b51ae3cb650a2870fae1b6705404bb155289e97be7e9a54961a9effb959
-COMPONENT_ARCHIVE_URL= https://www.process-one.net/downloads/ejabberd/$(HUMAN_VERSION)/$(COMPONENT_ARCHIVE)
-COMPONENT_ARCHIVE_URL= https://github.com/processone/$(COMPONENT_NAME)/archive/refs/tags/$(HUMAN_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH= sha256:0d6e7f0d82d91cda89e2575d99a83507413da2ffde39b2151804947a2a0fa258
+COMPONENT_ARCHIVE_URL= https://static.process-one.net/ejabberd/downloads/$(COMPONENT_VERSION)/ejabberd-$(COMPONENT_VERSION).tar.gz
 COMPONENT_BUGDB= service/ejabberd
 COMPONENT_SUMMARY= ejabberd - Jabber/XMPP instant messaging server
 COMPONENT_DESCRIPTION= the ejabberd Jabber/XMPP instant messaging server
 COMPONENT_CLASSIFICATION= Web Services/Communications
 COMPONENT_LICENSE= GPL-2.0
 COMPONENT_LICENSE_FILE= $(COMPONENT_NAME).license
 
+TEST_TARGET= $(NO_TESTS)
 include $(WS_MAKE_RULES)/common.mk
 
 CXX      += $(CC_BITS)
-CFLAGS   += $(JPEG_CPPFLAGS) $(JPEG_CFLAGS) -I$(USRINCDIR)/gd2
-CXXFLAGS += $(JPEG_CPPFLAGS) $(JPEG_CXXFLAGS) -I$(USRINCDIR)/gd2
-CPPFLAGS += $(JPEG_CPPFLAGS) -I$(USRINCDIR)/gd2
-LDFLAGS  += $(JPEG_LDFLAGS) -lnsl -lsocket
+CFLAGS   += $(JPEG_CPPFLAGS) $(JPEG_CFLAGS) -I$(USRINCDIR)/gd2 -finline-functions -funroll-loops
+CXXFLAGS += $(JPEG_CPPFLAGS) $(JPEG_CXXFLAGS) -I$(USRINCDIR)/gd2 -finline-functions -funroll-loops -fno-exceptions -fno-rtti
```

Comment:
Could you please add a note why these two new options are needed?  Also, please add them as a separate `CFLAGS += ` lines to keep these lines short.

Thank you.

Reply:
> erm, well, i can separate them but i can't explain what these options do or why it's usually good to have them in a comment in 2 or 3 sentences. somebody who doesn't know these at all has to read a little more about them than a brief comment. i assumed that everyone working on a repo like this is familiar with compilers. it's not like inlining or unrolling loops is something exotic.

I'm very sorry to see you cannot explain why you want/need these options.  OTOH, I believe these options are something that deserves explanation so here is my view:

After digging a bit into the `no-rtti` stuff it looks like that its use could be dangerous and cause surprises.  [Here](https://stackoverflow.com/questions/4486609/when-can-compiling-c-without-rtti-cause-problems) is a nice example.

My preliminary conclusion is that for general purpose environments like OpenIndiana where we prefer stability over the memory footprint (how much we would save with `-fno-rtti`?) the usage of `-fno-rtti` should be both well justified and limited to parts of the system where it really cannot cause any problems.

I'm sorry, but -1 for `-fno-rtti` from me here.

For `-funroll-loops` the `gcc` man page says:
```
This option makes code larger, and may or may not make it run faster.
```
Do you have any numbers justifying this option here?

Re `-finline-functions`: since we build with `-O2` by default (and I believe it is same for ejabberd too) this option should not be needed to be set explicitly.

Re `-fno-exceptions`: this is to disable some compile warnings.  Why do we need that?

Reply:
Same comment as in #16187 this software also has a small enough userbase that we can test it. I am unaware of any commercial user using it.